File: C++/Problems/Problems/problem_2028.cpp
#include "problem_2028.h"

bool problem_2028::test() {
    std::vector rolls { 3, 2, 4, 3 };
    int mean = 4;
    int n = 2;

    std::vector expected { 6, 6 };

    auto result = missingRolls(rolls, mean, n);

    print1DVector(result);

    return expected == result;
}

std::vector<int> problem_2028::missingRolls(std::vector<int> &rolls, int mean, int n) {
    int m = rolls.size();
    int req = (n + m) * mean;
    for (auto r : rolls)
        req -= r;
    req -= n;
    if (req < 0)
        return {};
    std::vector<int> result(n);
    for (int i = 0; i < n; ++i)
    {
        if (req <= 0)
            result[i] = 1;
        else
        {
            int big = std::min(req, 5);
            result[i] = big + 1;
            req -= big;
        }
    }

    return req <= 0 ? result : std::vector<int>();
}

File: Python/Problems/Problems/problem2028.py
import heapq
import math
import queue
import sys

from utils import print1DArray, print2DArray
from typing import List, Set, Optional
from collections import Counter, deque

from .problem import Problem

class Problem2028(Problem):
    def test(self) -> bool:
        rolls = [3, 2, 4, 3]
        mean = 4
        n = 2

        expected = [6, 6]

        result = self.missingRolls(rolls, mean, n)

        print1DArray(result)

        return expected == result

    def missingRolls(self, rolls: List[int], mean: int, n: int) -> List[int]:
        m = len(rolls)
        req = (n + m) * mean
        for r in rolls:
            req -= r

        req -= n
        if req < 0:
            return []
        result = []
        for i in range(n):
            if (req <= 0):
                result.append(1)
            else:
                big = min(req, 5)
                result.append(big + 1)
                req -= big

        return result if req <= 0 else []